packages/schematics/angular/application/files/module-files/src/main.ts.template

```diff
@@ -1,10 +1,16 @@
-<% if(!!viewEncapsulation) { %>import { ViewEncapsulation } from '@angular/core';
-<% }%>import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
-
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { AppModule } from './app/app.module';
+<% if(experimentalZoneless) { %> import { provideExperimentalZonelessChangeDetection from '@angular/core'; <% } %>
+
+const bootstrapOptions = {
+    <% if(!experimentalZoneless) { %>ngZoneEventCoalescing: true<% } else { %>
+    providers: [<% if(experimentalZoneless) { %>provideExperimentalZonelessChangeDetection()<% } %>];
+    <% } %>
+    <% if(!!viewEncapsulation) { %>
+      const { ViewEncapsulation } = await import('@angular/core');
+      defaultEncapsulation: ViewEncapsulation.<%= viewEncapsulation %>;
+    <% } %>
+};
 
-platformBrowserDynamic().bootstrapModule(AppModule, {
-  ngZoneEventCoalescing: true<% if(!!viewEncapsulation) { %>,
-  defaultEncapsulation: ViewEncapsulation.<%= viewEncapsulation %><% } %>
-})
+platformBrowserDynamic().bootstrapModule(AppModule, bootstrapOptions)
```

Comment:
Just noticed but, `providers` is not valid in this context, it needs to be added to the ngModule. 

https://github.com/angular/angular-cli/blob/d6dec2309f73417111b81e44f0a393d805f4739c/packages/schematics/angular/application/files/module-files/src/app/app.module.ts.template

```suggestion
platformBrowserDynamic().bootstrapModule(AppModule, {
  <% if(!experimentalZoneless) { %>ngZoneEventCoalescing: true, %><% if(!!viewEncapsulation) { %>
  defaultEncapsulation: ViewEncapsulation.<%= viewEncapsulation %><% } %>
})
```


See: https://angular.dev/guide/experimental/zoneless

Reply:
Looks like this has not been done. IE: the change in `app.module.ts.template`

Reply:
it seems to be ok or i received it wrong how to do the correction.

Reply:
Unless, `provideExperimentalZonelessChangeDetection` is added to app.module that ngModule based app will be configured to use zone.js, even when using `—experimental-zoneless`.